ament_cmake_gtest/cmake/ament_add_gtest.cmake

```diff
@@ -96,21 +96,31 @@ function(_ament_add_gtest target)
     set(ARG_WORKING_DIRECTORY "WORKING_DIRECTORY" "${ARG_WORKING_DIRECTORY}")
   endif()
 
-  ament_add_test(
-    "${target}"
-    COMMAND ${cmd}
-    OUTPUT_FILE "${CMAKE_BINARY_DIR}/ament_cmake_gtest/${target}.txt"
-    RESULT_FILE "${result_file}"
-    ${ARG_ENV}
-    ${ARG_APPEND_ENV}
-    ${ARG_APPEND_LIBRARY_DIRS}
-    ${ARG_TIMEOUT}
-    ${ARG_WORKING_DIRECTORY}
-  )
-  set_tests_properties(
-    "${target}"
-    PROPERTIES
-    REQUIRED_FILES "${executable}"
-    LABELS "gtest"
-  )
+  if(ARG_SKIP_TEST)
+    ament_add_test(
+      "${target}"
+      COMMAND ${cmd}
+      OUTPUT_FILE "${CMAKE_BINARY_DIR}/ament_cmake_gtest/${target}.txt"
+      RESULT_FILE "${result_file}"
+      SKIP_TEST
+    )
+  else()
+    ament_add_test(
+      "${target}"
+      COMMAND ${cmd}
+      OUTPUT_FILE "${CMAKE_BINARY_DIR}/ament_cmake_gtest/${target}.txt"
+      RESULT_FILE "${result_file}"
+      ${ARG_ENV}
+      ${ARG_APPEND_ENV}
+      ${ARG_APPEND_LIBRARY_DIRS}
+      ${ARG_TIMEOUT}
+      ${ARG_WORKING_DIRECTORY}
+    )
+    set_tests_properties(
+      "${target}"
+      PROPERTIES
+      REQUIRED_FILES "${executable}"
+      LABELS "gtest"
+    )
+  endif()
```

Comment:
Instead of duplicating the function call with all its arguments I would suggest to use a variable to either pass or not pass `SKIP_TEST`.